### PR TITLE
feat(extraction): add pydantic and experiment with vectorized indexing implementation

### LINKED ISSUE #9: Use conflator
### Is your feature request related to a problem? Please describe.

_No response_

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Organisation

_No response_

### PR DESCRIPTION
**Pull request by andreas-grafberger**
### Description

This refactor switches the regular extraction path to xarray vectorized indexing (sel/isel) instead of the current mask/ufunc flow. It should be less code and potentially up to ~2× faster based on simple tests, but I still want to run proper benchmarks before publishing this PR to verify the numbers.

Other changes:
- Pydantic config models end‑to‑end (single ExtractorConfig surface, CLI parses YAML → model → extractor)
- Expanded tests: descending‑lat grids, gribjump dedupe, dask‑backed inputs, validation edge cases, tolerance opt‑out

**Why**
   - Vectorized indexing avoids the sort‑order bug by construction.
   - The old mask/ufunc pipeline is more code and potentially slower.
   - Typed config gives us earlier, clearer validation errors.

**Tolerance / Safety**
This adds a coord‑matching tolerance (sane default still needs to be figure out). That acts as a threshold guard so stations outside the grid or too far from any grid point fail fast instead of silently mapping to a coarse/incorrect location. Users can disable this with tolerance: null if they want pure nearest‑neighbor behavior.

**Open / TODO**

- Run a full production‑scale benchmark before publishing the PR. The new path should be ~2× faster, but I want the numbers in hand.
- Review and possibly simplify tests
- Document that if we want to use a default for the tolerance that this will be a breaking change

### Contributor Declaration

By opening this pull request, I affirm the following:

* All authors agree to the [Contributor License Agreement](https://github.com/ecmwf/codex/blob/main/Legal/contributor_license_agreement.md).
* The code follows the project's coding standards.
* I have performed self-review and added comments where needed.
* I have added or updated tests to verify that my changes are effective and functional.
* I have run all existing tests and confirmed they pass.

### CODE REVIEW

File: src/hyve/config.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    x: str = "opt_x_index"
+    y: str = "opt_y_index"
+
+
+class CoordMapping(BaseModel):
+    """Column names for coordinate-based station mapping.
+
+    Notes
+    -----
+    ``tolerance`` is expressed in degrees and defaults to ~10 m at the equator
```
Comment (colonesej) [inline src/hyve/config.py]: It follows the grid coordinates units, no!? for instance, if the grid is projected, tolerance would be in meters.

**Review by colonesej — APPROVED** [review]
very nice, @andreas-grafberger

**Comment by colonesej** [comment]
@andreas-grafberger how close are we to closing this? Do you need any specific help?
